### DISCUSSION TITLE
Run ferrum in an arm64 machine with docker

### DISCUSSION THREAD

**Discussion by devsheva** [rubycdp/ferrum#446] [Q&A]
How can we run ferrum in a container relying on arm64 machine ? Afaik,  at the moment chrome doesn't provide an arm binary for chromium

**Comment by route** [comment]
Google provides chromium binary for arm64 and chrome for x86-64

**Reply by devsheva** [reply]
Do you have a link ? I tried a lot of stuff, without success. You can simply check https://download-chromium.appspot.com/ and find that there isn't a linux arm64 binary

**Reply by devsheva** [reply]
Also watch out this, maybe I'm wrong. https://github.com/GoogleChromeLabs/chrome-for-testing?tab=readme-ov-file#json-api-endpoints

**Comment by route** [comment]
You can install it using snap tool https://snapcraft.io/chromium that's the easy way

**Reply by devsheva** [reply]
I'll give it a try, thanks

**Reply by devsheva** [reply]
this is what i get when i try to run snap inside the container. It's a basic Ubuntu ARM64 image.

```
snap install chromium
error: cannot communicate with server: Post "http://localhost/v2/snaps/chromium": dial unix /run/snapd.socket: connect: no such file or directory
```

**Reply by route** [reply]
I don't know your setup, something wrong with network

**Reply by devsheva** [reply]
just pulling [ubuntu](https://hub.docker.com/_/ubuntu) image and running it, nothing more.

Also i found out this in puppeteer docs. https://pptr.dev/troubleshooting

`Chromium currently does not provide arm64 binaries for Linux. There are only binaries for Mac ARM.`

**Reply by route** [reply]
Well there's definitely a chromium binary on ubuntu:

```shell
snap info chromium

name:      chromium
summary:   Chromium web browser, open-source version of Chrome
publisher: Canonical✓
store-url: https://snapcraft.io/chromium
contact:   https://bugs.launchpad.net/ubuntu/+source/chromium-browser/+bugs?field.tag=snap
license:   Apache-2.0 AND BSD-3-Clause AND LGPL-2.0 AND LGPL-2.1 AND MIT AND MS-PL AND (GPL-2.0+ OR LGPL-2.1+ OR MPL-1.1)
description: |
  An open-source browser project that aims to build a safer, faster, and more
  stable way for all Internet users to experience the web.
snap-id: XKEcBqPM06H1Z7zGOdG5fbICuf8NWK5R
channels:
  latest/stable:    122.0.6261.57 2024-02-23 (2762) 168MB -
  latest/candidate: 122.0.6261.69 2024-02-24 (2765) 168MB -
  latest/beta:      122.0.6261.57 2024-02-23 (2763) 168MB -
  latest/edge:      123.0.6286.0  2024-02-10 (2756) 168MB -
```
